test/html/test.ml

```diff
@@ -1,47 +1,124 @@
 open Compat
+open Printf
 
-let cases_directory = "cases"
-let expect_directory = "expect"
-let build_directory = "_scratch"
-let test_package = "test_package"
-let odoc = "../../src/odoc/bin/main.exe"
-let test_root = "test/html"
+(* Utils *)
 
 let (//) = Filename.concat
 
-let contains_actual = build_directory // "actual"
-let contains_expected = build_directory // "expected"
-
 let command label =
   Printf.ksprintf (fun s ->
     let exit_code = Sys.command s in
     if exit_code <> 0 then
       Alcotest.failf "'%s' exited with %i" label exit_code)
 
-let running_in_travis = match Sys.getenv "TRAVIS" with
-  | "true" -> true
-  | _ -> false
-  | exception Not_found -> false
-
-let found_tidy = match Sys.command "which tidy > /dev/null" with
-  | 0 -> true
-  | _ -> false
-
-let cases = [
-  "val.mli";
-  "markup.mli";
-  "section.mli";
-  "module.mli";
-  "interlude.mli";
-  "include.mli";
-  "mld.mld";
-  "type.mli";
-  "external.mli";
-  "functor.mli";
-  "class.mli";
-  "stop.mli";
-]
-
+(* Filename.extension is only available on 4.04. *)
+module Filename = struct
+  include Filename
+
+  let extension filename =
+    let dot_index = String.rindex filename '.' in
+    String.sub filename dot_index (String.length filename - dot_index)
+end
+
+
+(* Testing environment *)
+
+module Env = struct
+  let package = "test_package"
```

Comment:
I guess that could happen. I'd prefer to clean the `_scratch` directory instead of generating new copies of the HTML results. This could also be addressed in a separate PR.

 @aantron what are your thoughts on this?

Reply:
Cleaning `_scratch` would also work for me.

+1 to doing it on a separate PR.

Reply:
Cleaning `_scratch`, and doing it in a separate PR, both sound good to me.